src/tests/suite/daos_nvme_recovery.c

```diff
@@ -254,11 +254,119 @@ nvme_test_verify_device_stats(void **state)
 	D_FREE(devices);
 }
 
+/*
+ * Verify blobstore state transitions from NORMAL->OUT after device is marked
+ * as "FAULTY" by querying the internal blobstore device state by calling the
+ * daos_mgmt_get_bs_state() C API.
+ */
+static void
+nvme_test_get_blobstore_state(void **state)
+{
+	test_arg_t	*arg = *state;
+	device_list	*devices = NULL;
+	daos_obj_id_t	 oid;
+	struct ioreq	 req;
+	char		 dkey[DTS_KEY_LEN] = { 0 };
+	char		 akey[DTS_KEY_LEN] = { 0 };
+	int		 obj_class, key_nr = 10;
+	int		 rank = 0, tgt_idx = 0;
+	int		 blobstore_state;
+	int		 i, j;
+	int		 ndisks;
+	int		 rc;
+
+	if (!is_nvme_enabled(arg)) {
+		print_message("NVMe isn't enabled.\n");
+		skip();
+	}
+
+	/**
+	 * Get the total number of NVMe devices from all the servers.
+	 */
+	rc = dmg_storage_device_list(dmg_config_file, &ndisks, NULL);
+	assert_int_equal(rc, 0);
+	print_message("Total Disks = %d\n", ndisks);
+
+	/**
+	 * Get the Device info of all NVMe devices.
+	 */
+	D_ALLOC_ARRAY(devices, ndisks);
+	rc = dmg_storage_device_list(dmg_config_file, NULL, devices);
+	assert_int_equal(rc, 0);
+	for (i = 0; i < ndisks; i++)
+		print_message("Rank=%d UUID=%s state=%s host=%s\n",
+			      devices[i].rank, DP_UUID(devices[i].device_id),
+			devices[i].state, devices[i].host);
+
+
+	/**
+	 * Set the object class and generate data on objects.
+	 */
+	if (arg->pool.pool_info.pi_nnodes < 2)
+		obj_class = DAOS_OC_R1S_SPEC_RANK;
+	else
+		obj_class = DAOS_OC_R2S_SPEC_RANK;
+
+	oid = dts_oid_gen(obj_class, 0, arg->myrank);
+	oid = dts_oid_set_rank(oid, rank);
+	oid = dts_oid_set_tgt(oid, tgt_idx);
+	ioreq_init(&req, arg->coh, oid, DAOS_IOD_ARRAY, arg);
+
+	print_message("Generating data on obj "DF_OID"...\n", DP_OID(oid));
+	for (i = 0; i < key_nr; i++) {
+		dts_key_gen(dkey, DTS_KEY_LEN, "dkey");
+		for (j = 0; j < key_nr; j++) {
+			dts_key_gen(akey, DTS_KEY_LEN, "akey");
+			insert_single(dkey, akey, 0, "data", strlen("data") + 1,
+				      DAOS_TX_NONE, &req);
+		}
+	}
+	ioreq_fini(&req);
+
+	/**
+	 * Verify blobstore of first device returned is in "NORMAL" state
+	 * before setting to faulty.
+	 */
+	rc = daos_mgmt_get_bs_state(arg->group, devices[0].device_id,
+				    &blobstore_state, NULL /*ev*/);
+	assert_int_equal(rc, 0);
```

Comment:
Getting below error for some of the different count of targets

**16 Target--1 Disk- Working**
1 Target--1 Disk- Not Working
8 Targets--3 Disks- Not Working

```
[ RUN      ] NVMe Recovery 3: Verify blobstore device state NORMAL->OUT transition
Total Disks = 4
Rank=3 UUID=206b246b-276b-4a48-8e74-b167485471a8 state="NORMAL" host=wolf-155
Rank=1 UUID=7ddc2abd-3c4b-43af-b973-50cc69424a25 state="NORMAL" host=wolf-155
Rank=0 UUID=f0fa0541-de6c-41b5-8810-4f35e2495870 state="NORMAL" host=wolf-154
Rank=2 UUID=82fc8176-5dbd-49b0-a8fd-cfa2fa86d659 state="NORMAL" host=wolf-154
Generating data on obj 1152921586211225616.1...
[  ERROR   ] --- 0xfffffffffffffc13 != 0
[   LINE   ] --- src/tests/suite/daos_nvme_recovery.c:332: error: Failure!
```

```
[ RUN      ] NVMe Recovery 3: Verify blobstore device state NORMAL->OUT transition
Total Disks = 12
Rank=3 UUID=0315da8f-0289-43d8-a253-01ea3d6fd6fb state="NORMAL" host=wolf-155
Rank=3 UUID=480b9ab3-3711-4494-814b-ac90c84cf091 state="NORMAL" host=wolf-155
Rank=3 UUID=72bf7916-fb27-4ecb-a12b-dc09825eaf25 state="NORMAL" host=wolf-155
Rank=1 UUID=10727e71-5cf1-483d-a946-3f742295b8da state="NORMAL" host=wolf-155
Rank=1 UUID=1e1463a3-9d9f-4821-9a39-2287d86654c4 state="NORMAL" host=wolf-155
Rank=1 UUID=3a37a003-379e-4c43-abf8-d091009555a6 state="NORMAL" host=wolf-155
Rank=0 UUID=4ce27cab-ca9a-400a-be09-21dea02f0900 state="NORMAL" host=wolf-154
Rank=0 UUID=dde3d900-4f8f-4a6f-8e26-e43591171a08 state="NORMAL" host=wolf-154
Rank=0 UUID=eff0943d-010d-4eb0-89f8-245d6e737058 state="NORMAL" host=wolf-154
Rank=2 UUID=4ef4d1fe-b8d3-4c9a-a28d-7464f6718caf state="NORMAL" host=wolf-154
Rank=2 UUID=d1b62bda-b80a-4ca3-8c04-3b9bcceb4815 state="NORMAL" host=wolf-154
Rank=2 UUID=f128bb7a-2e47-4b82-ae98-f71f4dcdcbd0 state="NORMAL" host=wolf-154
Generating data on obj 1152921586211225645.1...
[  ERROR   ] --- 0xfffffffffffffc13 != 0
```

Reply:
worked with Samir and could not reproduce this issue again, might be a test yaml configuration issue

Reply:
Yes it was something with the test yaml configuration. I retested the above configuration and it's working fine.